### PR TITLE
feat: implmenet a use defaults system WIP fixes #1607 

### LINKED ISSUE #1607: Global options
A requirement of the library is to have a mechanism for creating "global options". Such as setting the default for props. 

How exactly this would work idk yet.

But for instance this could be applied to having the nuxt module convert BLinks to NuxtLink (instead of RouterLink) by default, and other Nuxt specific options

### PR DESCRIPTION
**Pull request by VividLemon**
# Describe the PR

A clear and concise description of what the pull request does.

## Small replication

A small replication or video walkthrough can help demonstrate the changes made. This is optional, but can help observe the intended changes. A mentioned issue that contains a replication also works.

## PR checklist

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix :bug: - `fix(...)`
- [ ] Feature - `feat(...)`
- [ ] ARIA accessibility - `fix(...)`
- [ ] Documentation update - `docs(...)`
- [ ] Other (please describe)

**The PR fulfills these requirements:**

- [ ] Pull request title and all commits follow the [**Conventional Commits**](https://www.conventionalcommits.org/) convention or has an [**override**](https://github.com/googleapis/release-please#how-can-i-fix-release-notes) in this pull request body **This is very important, as the `CHANGELOG` is generated from these messages, and determines the next version type. Pull requests that do not follow conventional commits or do not have an override will be denied**

### CODE REVIEW

File: packages/bootstrap-vue-next/src/types/BootstrapVueOptions.ts [hunk 1]
```diff
@@ -63,5 +71,6 @@ export interface BootstrapVueOptions {
      * @default true
      */
     toast?: boolean
+    components?: Partial<ComponentProps>
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/types/BootstrapVueOptions.ts:74]: Duplicate key `components`

**Review by VividLemon — COMMENTED** [review]
Remember to revert main.ts

**Comment by xvaara** [comment]
Just heading to sleep. 5min look seems nice. I'll have time next week to look at this and other stuff.

Should we add the handeling of modelValue to useDefaults? pass emit to it, revert back to vueuse style handeling of it... Just an idea, not really sure about it.

**Comment by VividLemon** [comment]
> Should we add the handeling of modelValue to useDefaults? pass emit to it, revert back to vueuse style handeling of it... Just an idea, not really sure about it.

There are some examples of how its used. Its a bit strange code, all boilerplate. But its just a computed setter.

**Comment by VividLemon** [comment]
@xvaara Something that I didn't think about is the fact that we don't know whether or not the modelvalue is bound. This causes a problem:

```ts
const props = useDefaults<typeof _props>(_props, 'BAccordion')
const _modelValue = defineModel<string | undefined>({
  default: undefined,
})
const modelValue = computed({
  get: () => props.modelValue,
  set: (v) => {
    _modelValue.value = v
  },
})
```
When we bind to props.modelValue when it's using the "passive" version of the modelValue, it's then bound to the wrong value. 

```html
<BAccordion /> 
```
This won't work, because internally its supposed to be using the ref. However, since we bounded to the props, it fails. 

I'm not sure if there's a way to detect this (I don't think there is). Meaning, I think we may just have to refuse global options for modelValues

File: apps/docs/src/docs/configurations/global-options.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    BSpinner: {
+      type: 'grow',
+    },
+  },
+})
+```
+
+In the above example, each key of the components object corresponds to a Vue component name, and the value is an object that defines the default prop values for that component.
+
+The special key global can be used to set a default value for a prop across all components. For example, if you set type: 'grow' under global, all components that have a type prop will default to 'grow', regardless of whether the component's prop definition accepts it.
```
Comment (VividLemon) [inline apps/docs/src/docs/configurations/global-options.md:26]:
```suggestion
The special key `global` can be used to set a default value for a prop across all components. For example, if you set type: 'grow' under global, all components that have a type prop will default to 'grow', regardless of whether the component's prop definition accepts it.
```

File: packages/bootstrap-vue-next/src/components/BAccordion/BAccordion.vue [hunk 3]
```diff
@@ -21,6 +21,8 @@ defineSlots<{
   default?: (props: Record<string, never>) => any
 }>()
 
+const props = useDefaults<typeof _props>(_props, 'BAccordion')
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BAccordion/BAccordion.vue:24]:
```suggestion
const props = useDefaults(_props, 'BAccordion')
```

**Comment by VividLemon** [comment]
feat: implement a use defaults system fixes #1607

feat: implement global alias system fixes #1753

refactor: add bvn:: prefix to symbol keys

refactor: fix typo in 'BCardImageProps' change to 'BCardImgProps'

feat(BFormInput)!: move props lazy, trim & number to v-model modifier. Modifiers should behave more like native Vue modifiers (number modifier with unparseable value returns value as is)

feat(BFormInput): type number performs the same as modifier

feat(BCarouselSlide): allow individual interval for slides

chore: update deps to fix some issues (nuxt useId, vue-tsc etc)

docs: global options, aliases

feat(nuxt): automatically globally set teleportTo to `#teleports`

feat(BreakpointProps)!: strongly type breakpoints. Instead of weak string | number => 1,2,3...'1','2','3'...

refactor(BRow): use script setup instead of options api

refactor(BCol): use script setup instead of options api

**Pull request merged**